spec/integ/matrix-to-irc.spec.js

```diff
@@ -548,6 +550,47 @@ describe("Matrix-to-IRC message bridging with media URL and drop time", function
         expect(said).toBe(false);
     }));
 
+    it("should NOT bridge old matrix messages younger than the drop time on receive, which " +
+    "then go over the drop time whilst processing", test.coroutine(function*() {
+        const tBody = "Hello world";
+        const testUser2 = {
+            id: "@tester:wibble",
+            nick: "M-tester"
+        };
+        jasmine.clock().mockDate();
+
+        let said = false;
+        env.ircMock._whenClient(roomMapping.server, testUser2.nick, "say",
+        function(client, channel, text) {
+            said = true;
+        });
+
+        let connected = false;
+        env.ircMock._whenClient(roomMapping.server, testUser2.nick, "connect",
+        function(client, cb) {
+            jasmine.clock().tick(20 * 1000); // advance 20s to take it over the drop time
```

Comment:
I would make it obvious that "drop time" is actually `dropMatrixMessagesAfterSecs`